Introduce polling for live update of messages and notifications

sonalkr132 opened the pull request:
- [x] update to rails 4.2.5.1
- [x] live messages
- [x] live notification
- [x] tests

sonalkr132 committed:
Update rails to 4.2.5.1 and other gems

..5.1 part cause of security vulnerable in rails 4.2, check out following links:
https://groups.google.com/forum/#!topic/rubyonrails-security/cawsWcQ6c8g
https://groups.google.com/forum/#!topic/rubyonrails-security/ANv0HDHEC3k and others.
We are updating rails cause we will be introducing live messages through actioncable
of rails 5.
In 4.2 following changes were needed:
  * serve_static_assets was renamed to serve_static_files and will be removed in rails 5
  * change of true to cascade cause http://edgeguides.rubyonrails.org/4_2_release_notes.html#active-record-notable-changes

sonalkr132 committed:
Fix added for breaking change of rails 4.5.2.1 for rspec

sonalkr132 committed:
Downgrade shoulda-matchers from 3.1 to 2.8

3.1 seems to be incompatible with rails 4.2.5.1
Error: undefined method `validate_length_of' for #<RSpec::ExampleGroups ..>

sonalkr132 committed:
Fix URI::InvalidURIError due to missing forward slash

sonalkr132 commented:
Guide followed for update to rails 4.2.5.1: http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-4-1-to-rails-4-2

`web-console` and `responders` was not added cause we don't really need it. We might consider introducing `spring`, it has been included by default since rails 4.1, however rails 5 seems to have issues it for now.

One of the crucial updates is update of devise to 3.5.5 due to security vulnerable. You can read about it [here](http://blog.plataformatec.com.br/2016/01/improve-remember-me-cookie-expiration-in-devise/).

sonalkr132 committed:
Fix code style errors due to update of rubocop

sonalkr132 committed:
Rename haml-lint to haml_lint and fix no newline at end of file error

sonalkr132 commented:
Phew! Finally travis is green. Drop in test coverage is due to update of `simplecov`. And they are right, some of our files have 0% which weren't included in test coverage previously, they are included now.
However, I won't be working on test coverage in this PR.

sonalkr132 committed:
Introduce live comments with polling

Polling delay should be bumped up in production.
Some data attr were introducted for ajax request during polling.

sonalkr132 committed:
Fix indentation warning and make routes more general

sonalkr132 committed:
Introduce live notification with polling

sonalkr132 committed:
Add authorization and test for comment index

sonalkr132 committed:
Test for notification count action

Alwahsh commented on app/assets/javascripts/comments.js.coffee:
```diff
@@ -0,0 +1,17 @@
+@CommentPoller =
+	poll: ->
+		setTimeout @request, 30000
```
How about putting that number in some variable for configuration? I don't think it should be hard coded.

sonalkr132 replied:
Thanks @Alwahsh . I have made the required change.

sonalkr132 committed:
Move polling time to config variable

Set your polling time in 0_glitterconfig.rb:
Glitter::Application.config.polling_time = <milisecond>
